src/OpenLoco/Vehicles/Routing.cpp

```diff
@@ -0,0 +1,670 @@
+
+    // Reverse direction map?
+    static loco_global<uint8_t[16], 0x00503CAC> _503CAC;
+    static loco_global<Map::Pos2[16], 0x00503C6C> _503C6C;
+    static loco_global<FilterFunction, 0x01135F0E> _filterFunction;
+    static loco_global<uint32_t, 0x01135F0A> _1135F0A;
+    static loco_global<uint16_t, 0x01135FA6> _1135FA6;
+    static loco_global<TransformFunction, 0x01135F12> _transformFunction;
+    static loco_global<uint8_t, 0x01136085> _1136085;
+    static loco_global<LocationOfInterestHashMap*, 0x01135F06> _1135F06;
+    static loco_global<uint8_t[2], 0x0113601A> _113601A;
+
+    // 0x0048963F
+    uint8_t sub_48963F(const Map::Pos3& loc, const TrackAndDirection::_TrackAndDirection trackAndDirection, const uint8_t trackType, uint32_t flags)
+    {
+        const auto unk1 = flags & 0xFFFF;
+        auto trackStart = loc;
+        if (trackAndDirection.isReversed())
+        {
+            auto& trackSize = Map::TrackData::getUnkTrack(trackAndDirection._data);
+            trackStart += trackSize.pos;
+            if (trackSize.rotationEnd < 12)
+            {
+                trackStart -= Map::Pos3{ _503C6C[trackSize.rotationEnd], 0 };
+            }
+            flags ^= (1ULL << 31);
+        }
+
+        auto& trackPieces = Map::TrackData::getTrackPiece(trackAndDirection.id());
+        for (auto& trackPiece : trackPieces)
+        {
+            auto signalLoc = trackStart + Map::Pos3{ Math::Vector::rotate(Map::Pos2{ trackPiece.x, trackPiece.y }, trackAndDirection.cardinalDirection()), 0 };
+            signalLoc.z += trackPiece.z;
+            auto tile = Map::TileManager::get(signalLoc);
+            Map::TrackElement* foundTrack = nullptr;
+            for (auto& el : tile)
+            {
+                if (el.baseZ() != signalLoc.z / 4)
+                {
+                    continue;
+                }
+
+                auto* elTrack = el.as<TrackElement>();
+                if (elTrack == nullptr)
+                {
+                    continue;
+                }
+
+                if (!elTrack->hasSignal())
+                {
+                    continue;
+                }
+
+                if (elTrack->unkDirection() != trackAndDirection.cardinalDirection())
+                {
+                    continue;
+                }
+
+                if (elTrack->sequenceIndex() != trackPiece.index)
+                {
+                    continue;
+                }
+
+                if (elTrack->trackObjectId() != trackType)
+                {
+                    continue;
+                }
+
+                if (elTrack->trackId() != trackAndDirection.id())
+                {
+                    continue;
+                }
+                foundTrack = elTrack;
+                break;
+            }
+
+            if (foundTrack == nullptr)
+            {
+                if (unk1 == 10)
+                {
+                    return 0;
+                }
+                return loc.x; // Odd???
```

Comment:
What are callers of this function checking afterwards? Is it CF or a general purpose register?

Reply:
There are lots of callers. They can be split into two main categories those that expect eax to have information and those that don't. I am of the impression that those that expect eax to have information never gets to this statement

Reply:
That is what I thought, it might just be the data left over in eax, but the caller expects void return anyway.

Reply:
What i think i will do in a follow on is split this function into 2+ functions as the callers have pretty obvious different use cases. But as this function is from a previous PR I will leave it for now.